### PR TITLE
fix(disrupt_mgmt_repair_cli): stop task in case of timeout

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
In case the manager repair task does not finish by the time it reaches timeout, We will stop the task at the end of the nemesis, so it won't continue past it.

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by fgelcer — DISMISSED** [review]
LGTM

**Comment by fgelcer** [comment]
ping @yarongilor 
@fruch @roydahan , can you please review and merge?

**Comment by fgelcer** [comment]
ping @scylladb/qa-maintainers @ShlomiBalalis

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -2668,9 +2668,14 @@ def disrupt_mgmt_repair_cli(self):
         mgr_cluster = self.cluster.get_cluster_manager()
         mgr_task = mgr_cluster.create_repair_task()
         task_final_status = mgr_task.wait_and_get_final_status(timeout=86400)  # timeout is 24 hours
-        assert task_final_status == TaskStatus.DONE,\
-            f'Task: {mgr_task.id} final status is: {str(mgr_task.status)}.\nTask progress string: ' \
-            f'{mgr_task.progress_string(parse_table_res=False, is_verify_errorless_result=True).stdout}'
+        try:
```
Comment (yarongilor) [inline sdcm/nemesis.py:2671]: why not replacing try-except with if-else?
like: `if task_final_status == TaskStatus.DONE:`
```
else: 
            if task_final_status not in (TaskStatus.ERROR, TaskStatus.ERROR_FINAL):
                mgr_task.stop()
            raise
```

**Comment by fgelcer** [comment]
ping @ShlomiBalalis

**Review by yarongilor — APPROVED** [review]
LGTM

File: sdcm/nemesis.py [hunk 2]
```diff
@@ -2668,9 +2669,14 @@ def disrupt_mgmt_repair_cli(self):
...
         task_final_status = mgr_task.wait_and_get_final_status(timeout=86400)  # timeout is 24 hours
-        assert task_final_status == TaskStatus.DONE,\
-            f'Task: {mgr_task.id} final status is: {str(mgr_task.status)}.\nTask progress string: ' \
-            f'{mgr_task.progress_string(parse_table_res=False, is_verify_errorless_result=True).stdout}'
+        if task_final_status != TaskStatus.DONE:
+            progress_full_string = mgr_task.progress_string(
+                parse_table_res=False, is_verify_errorless_result=True).stdout
+            if task_final_status != TaskStatus.ERROR_FINAL:
+                mgr_task.stop()
+            raise ScyllaManagerError(
```
Comment (fgelcer) [inline sdcm/nemesis.py:2677]: BTW, IMHO same must be done for backup

**Review by fgelcer — APPROVED** [review]
LGTM

File: sdcm/nemesis.py (hunk 2 above)
Comment (ShlomiBalalis) [reply sdcm/nemesis.py:2677]: You mean stopping the task at the end? It's already a part of the backup nemesis

**Comment by ShlomiBalalis** [comment]
@scylladb/qa-maintainers Ready to be merged

**Comment by fgelcer** [comment]
> @scylladb/qa-maintainers Ready to be merged

ping @scylladb/qa-maintainers

**Pull request merged**